src/mscorlib/src/System/UnSafeCharBuffer.cs

```diff
@@ -29,25 +29,43 @@ unsafe internal struct UnSafeCharBuffer{
             m_totalSize = bufferSize;    
             m_length = 0;
         }
-    
+
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        public void AppendChar( char charToAppend ) {
+            if( charToAppend == '\0' ) {
+                return;
+            }
+
+            if ( (m_totalSize - m_length) < 1 ) {
+                throw new IndexOutOfRangeException();
+            }
+
+            byte charToAppendByte = (byte) charToAppend;
+
+            Buffer.Memcpy( (byte*) (m_buffer + m_length), (byte *) charToAppendByte, sizeof(char) );
```

Comment:
Is a call to `Buffer.Memcpy` really necessary here? Shouldn't you do something like

```cs
m_buffer[m_length++] = charToAppend;
```